src/aggregators/ssajump.jl

```diff
@@ -222,6 +224,33 @@ Notes:
     return jidx
 end
 
+"""
+    linear_search(iterator, r)
+
+Perform linear search for `r` over iterator. Output index j s.t. sum(array[1:j-1])
+< r <= sum(array[1:j]), where array=collect(iterator)
+
+Notes:
+- The iterator must have all positive numbers
+- Returns index zero if the search is unsuccessful. Assumes this corresponds to
+  the case of an infinite next reaction time and so the jump index does not
+  matter.
+"""
+function linear_search(iterator, r)
```

Comment:
linear search over an iterator instead of an array.

Reply:
Also, why drop the inline?

Reply:
Looks like there is no difference noticeable. I tested Direct on gene expression and found no difference in allocations or performance